apps/tx-builder/src/components/Builder.tsx

```diff
@@ -281,6 +286,48 @@ export const Builder = ({
     }
   }, [transactions]);
 
+  const renderInput = (input: any, index: number) => {
```

Comment:
I think we'd benefit more if this would be a separate component; IMO this `Builder.tsx` component is already long enough

Reply:
I agree with you but I decide to make the simplest change because I think is better to create separate components per field type (more or less like the AddressContractField one) with validation included ... and more independent because If you take a look to the AddressContractField there are props sent to it totally dependent on the Builder like the index position or the input itself, so no very decoupled al the end.

As we are going to have a ticket for implementing validation, I think it will be the right place to refactor this loop creating the individual decoupled components managing rendering and validation and being controlled by the builder.
 
So I felt that I was going to create a component that at the end we are going to change again shortly for something more elaborated